pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,269 @@
+        categorical distribution:
+
+        $$P(y=c|x_i) = Cat(\pi_1, \dots, \pi_C)$$
+
+        where $\pi_i$ is the probability that a vector belongs to category $i$.
+        
+        Priors
+        ------
+
+        We assume that the $\pi_i$ follow a Dirichlet distribution:
+
+        $$\pi \sim Dirichlet(\alpha)$$
+
+        with hyperparameter $\alpha = [1, .., 1]$. The $\mu_{jc}$ are sampled from
+        a Normal distribution centred on $0$ with variance $100$, and the $\sigma_{jc}$
+        are sampled from a HalfNormal distribuion of variance $100$:
+
+        $$\mu_{jc} \sim Normal(0, 100)$$
+
+        $$\sigma_{jc} \sim HalfNormal(100)$$
+
+        Note that the Gaussian Naive Bayes model is equivalent to a Gaussian
+        mixture with a diagonal covariance [1].
+
+        Returns
+        -------
+        A PyMC3 model
+
+        [1]: Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+
+        """
+        
+        # The data
+        X = theano.shared(np.zeros((self.n_samples, self.n_features)))
+        y = theano.shared(np.zeros(self.n_samples))
+
+        self.shared_vars = {
+            'model_input': X,
+            'model_output': y
+        }
+        
+        model = pm.Model()
+        with model:
+            # Priors
+            alpha = np.ones(self.n_classes)
+            pi = pm.Dirichlet('pi', alpha, shape=self.n_classes)
+            mu = pm.Normal('mu', 0, 100, shape=(self.n_classes, self.n_features))
+            sigma = pm.HalfNormal('sigma', 100, shape=(self.n_classes, self.n_features))
+
+            # Assign classes to data points
+            z = pm.Categorical('z', pi, shape=self.n_samples, observed=y)
+
+            # The components are independant and normally distributed
+            xi = pm.Normal('xi', mu[z], sigma[z], observed=X)
+
+        return model
+
+
+    def fit(self, X, y, inference_type='advi', minibatch_size=None, inference_args=None):
+        """ Train the Naive Bayes model.
+
+        Parameters
+        ----------
+        X (np.array): Numpy array of shape [n_samples, n_features] containing the 
+                      data points.
+        y (np.array): Numpy array of shape [n_samples,] containing the category 
+                      of the data points.
+        inference_type (string): Specifies which inference method to call. Default is
+                                 'advi'. Currently, only 'advi' and 'nuts'are
+                                 implemented.
+        minibatch_size (int): Number of samples to include in each minibatch for
+                              ADVI. Defaults to None so minibatch is not run by
+                              default.
+        inference_args (dict): Arguments to be passed to the inference methods. Check
+                               the PyMC3 documentation.
+        
+        Returns
+        -------
+        The current instance of the model.
+
+        """
+        self.n_samples, self.n_features = X.shape 
```

Comment:
I was trying to follow Scikit-learn's naming conventions (see the code for [k-means](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cluster/k_means_.py) for instance). I personally find they make sense. I'm aware it's a subjective notion, but I think it would be wise to adopt the same conventions as sk-learn here.